src/shared/mount-util.c

```diff
@@ -163,7 +168,13 @@ static int get_mount_flags(
         return 0;
 
 fallback:
-        if (statvfs(path, &buf) < 0)
+        fd = open(path, O_PATH|O_CLOEXEC|O_NOFOLLOW);
+        if (fd < 0)
+                return -errno;
+
+        xsprintf(fd_path, "/proc/self/fd/%i", fd);
+
+        if (statvfs(fd_path, &buf) < 0)
```

Comment:
Wouldn't it be nicer to use fstatvfs here?

Reply:
hmm, i assumed fstatvfs() wouldn't work on O_PATH. But as it turns out it does since kernel 3.12. Our current minimum kernel is 3.13, hence let's rely on it.